### ISSUE TITLE
Xcode 10: unable to open dependencies file when archiving a project

### ISSUE THREAD

**Issue by Amurmurmur** [facebook/react-native#21137]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment
Run `react-native info` in your terminal and paste its contents here.
```
React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.6
      CPU: x64 Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
      Memory: 443.61 MB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.10.0 - /usr/local/bin/node
      Yarn: 1.9.4 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.0, macOS 10.14, tvOS 12.0, watchOS 5.0
      Android SDK:
        Build Tools: 23.0.1, 23.0.2, 23.0.3, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 27.0.0, 27.0.1, 27.0.2, 27.0.3
        API Levels: 22, 23, 24, 25, 26, 27
    IDEs:
      Android Studio: 3.1 AI-173.4907809
      Xcode: 10.0/10A254a - /usr/bin/xcodebuild
    npmPackages:
      @storybook/react-native: ^4.0.0-alpha.12 => 4.0.0-alpha.21
      react: 16.5.1 => 16.5.1
      react-native: ^0.57.0 => 0.57.0
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```

## Description
Describe your issue in detail. Include screenshots if needed. If this is a regression, let us know.
Upgraded to the latest XCode 10 GM Seed.
Upgraded RN to 0.57.0
Everything used to compile just fine just before the upgrade of the Xcode and RN
I have Pods in my project.
I have 3 build types: Debug, Staging and Release.
When trying to archive the app the following error gets thrown:
```
▸ Compiling project_name_vers.c

❌  error: unable to open dependencies file (/Users/amurmurmur/Library/Developer/Xcode/DerivedData/project_name-euciyyahtznlbecqdxaahvogieoa/Build/Intermediates.noindex/ArchiveIntermediates/project_name/IntermediateBuildFilesPath/project_name.build/Staging-iphoneos/project_name.build/Objects-normal/armv7/project_name_vers.d)Command CompileC emitted errors but did not return a nonzero exit code to indicate failure


▸ Compiling main.m
** ARCHIVE FAILED **


The following commands produced analyzer issues:
AnalyzeShallow /project_name/node_modules/react-native-fast-image/ios/Vendor/SDWebImage/Vendors/libwebp/src/enc/analysis_enc.c normal arm64
AnalyzeShallow /project_name/node_modules/react-native-fast-image/ios/Vendor/SDWebImage/Vendors/FLAnimatedImage/FLAnimatedImage/FLAnimatedImageView.m normal arm64
AnalyzeShallow /project_name/node_modules/react-native-fast-image/ios/Vendor/SDWebImage/Vendors/FLAnimatedImage/FLAnimatedImage/FLAnimatedImage.m normal arm64
(3 commands with analyzer issues)
[22:28:14]: Exit status: 65
```

Hopefully you can help me, already wasted 2 days trying to figure it out with no luck :(

## Reproducible Demo
Let us know how to reproduce the issue. Include a code sample, share a project, or     share an app that reproduces the issue using https://snack.expo.io/. Please follow the guidelines for providing a MCVE: https://stackoverflow.com/help/mcve

**Comment by kelset** [comment]
Seems to me this is related to https://github.com/facebook/react-native/issues/19573

**Comment by Amurmurmur** [comment]
Resolved with the latest RN update 0.57.5

**Comment by MaheshNandam** [comment]
RN 0.57.7 and XCode 10.2

if I'm trying to run the app into 12.2 version simulators or devices, then app got crashed with
Thread 1: signal SIGABRT

ISSUE LINE -> 0x102eda2f8<+8>: b.lo 0x102eda310 ; <+32>

if I'm trying to run into 12.1 version simulators or devices, it works fine.

With this (RN 0.57.1 and XCode 10.2) configuration, if I generate IPA, it's only working in 12.1 devices and when it comes to 12.2 devices, app getting the crash on the launch screen with Thread 1: signal SIGABRT.

Any suggestions or workarounds to solve this issue?

Thanks for that savior.
